### PR TITLE
Finished getFileTypeFromPath function

### PR DESCRIPTION
**Pull request by Manpreet-Bhatti**
Added extra FileType that accounts for config files whilst finishing the getFileTypeFromPath. Not sure if throw was correctly done -- please look into that.

### CODE REVIEW

File: src/utils/getFileTypeFromPath.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { FileType } from "../tree/types";
+
+export const getFileTypeFromPath = (path: string): FileType => {
```
Comment (ralph-dev) [inline src/utils/getFileTypeFromPath.ts:3]: needs JSDOC

File: src/utils/getFileTypeFromPath.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const count = (path.match(/\./g) || [])
+    .length; /* find the amount of periods assuming the path is relative and starts with "./" */
+
+  if (count == 3) {
```
Comment (ralph-dev) [inline src/utils/getFileTypeFromPath.ts:7]: no magic numbers....
const CONFIG_NUMBER_OF_DOTS = 3

Comment (Manpreet-Bhatti) [reply src/utils/getFileTypeFromPath.ts:7]: Changed `count` to`numberOfDots`

File: src/utils/getFileTypeFromPath.ts (hunk 1 above)
Comment (Manpreet-Bhatti) [reply src/utils/getFileTypeFromPath.ts:3]: `/**
 * @function
 * @param {string} path
 * @returns FileType}
 */`

Does this look good?

Comment (ralph-dev) [reply src/utils/getFileTypeFromPath.ts:10]: Where is the description? Push the new code and i will approve if it good or leave another comment

File: src/utils/getFileTypeFromPath.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { FileType } from "../tree/types";
+
+/**
+ * @function
+ * @param path string
```
Comment (ralph-dev) [inline src/utils/getFileTypeFromPath.ts:5]: not JSDOC format

File: src/utils/getFileTypeFromPath.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * @returns FileType
+ * @description using inputted relative GitHub path, find the FileType of the path depending on how many dots are found in the string
+ */
+
+export const getFileTypeFromPath = (path: string): FileType => {
+  // Count amount of periods assuming the path is relative and starts with "./"
+  const numberOfDots = (path.match(/\./g) || []).length;
+
+  if (numberOfDots == 3) {
```
Comment (ralph-dev) [inline src/utils/getFileTypeFromPath.ts:14]: still magic numbers

File: src/utils/getFileTypeFromPath.ts (hunk 1 above)
Comment (sahandfarahani) [reply src/utils/getFileTypeFromPath.ts:10]: should we use @description like Manpreet did, or can we put the description in front of the @function as well?

File: src/utils/getFileTypeFromPath.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  const folderNumberOfDots = 1;
+
+  if (numberOfDots == configNumberOfDots) {
+    // Length of 3 means 2 dots in the file name -- config file
+    return FileType.CONFIG_FILE;
+  } else if (numberOfDots == fileNumberOfDots) {
+    // Length of 2 means 1 dot in the file name -- regular file
+    return FileType.FILE;
+  } else if (numberOfDots == folderNumberOfDots) {
+    // Length of 1 means there are no file names in path -- folder */
```
Comment (sahandfarahani) [inline src/utils/getFileTypeFromPath.ts:25]: this line has a closed */ where other lines doesn't

File: src/tree/types.ts [hunk 6]
```diff
@@ -21,11 +21,12 @@ export enum FileType {
   FOLDER = 'FOLDER',
   FILE = 'FILE',
   HIDDEN_FILE = 'HIDDEN_FILE',
+  CONFIG_FILE = 'CONFIG_FILE',
 }
 
 export interface TreeCore {
```
Comment (sahandfarahani) [inline src/tree/types.ts:27]: I guess you don't need to export interface here

File: src/utils/test.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { expect } from 'chai';
+import { getFileTypeFromPath } from './getFileTypeFromPath';
+import 'mocha';
+
+describe('Finds file type of path', () =>{
+    it('should return FILE, CONFIG_FILE, or FOLDER', () => {
+        const result = getFileTypeFromPath('./src/tree/index.ts');
+        expect(result).to.equal('FILE');
```
Comment (ralph-dev) [inline src/utils/test.ts:8]: You need tests for each case (Folders, Files, Configs, Edge Cases)

File: src/utils/getFileTypeFromPath.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import { FileType } from '../tree/types';
+
+/** Outputs the file type of a certain path
+ * @param {string} path the path of a specific file or folder
+ * @returns {FileType} file type such as .FILE, .CONFIG_FILE, or .FOLDER
+ * @description using inputted relative GitHub path, get file type of path
+ */
+
+export const getFileTypeFromPath = (path: string): FileType => {
+  // Count amount of periods assuming the path is relative and starts with "./"
```
Comment (sahandfarahani) [inline src/utils/getFileTypeFromPath.ts:10]: I just noticed something. 
The path which the API produces doesn't start with "./"
So if the clickable link would be something like [components](./src/components) but the address we get from API as path is going to be "src/components"

Comment (ralph-dev) [reply src/utils/getFileTypeFromPath.ts:10]: See that's why you need to check :)
Becuase if you just get 
src -> that becomes [src](./src)

Comment (Manpreet-Bhatti) [reply src/utils/getFileTypeFromPath.ts]: Thank you, I've fixed this!

Comment (Manpreet-Bhatti) [reply src/utils/getFileTypeFromPath.ts]: Hey @ralph-dev, mind checking over the function file and test case file to see if it's ready for merge? I've updated them both and changed the structure of the function so it's more adaptive at catching invalid paths.

File: src/utils/getFileTypeFromPath.ts (hunk 1 above)
Comment (ralph-dev) [reply src/utils/getFileTypeFromPath.ts:9]: You should remove function and description and just write the description :)

Comment (Manpreet-Bhatti) [reply src/utils/getFileTypeFromPath.ts:9]: @ralph-dev remove the JSDoc description and just write a general comment destruction?

Comment (ralph-dev) [reply src/utils/getFileTypeFromPath.ts:8]: this part should clearly explain the function itself. Outputs the file type of a certain path. You don't need to repeat it twice.
